CMakeLists.txt

```diff
@@ -238,6 +238,15 @@ list(APPEND libs ${thirdparty_lib})
 
 add_subdirectory(library)
 
+if(ENABLE_TESTING OR ENABLE_PROGRAMS)
+    file(GLOB MBEDTLS_TEST_FILES ${CMAKE_CURRENT_SOURCE_DIR}/tests/src/*.c)
+    add_library(mbedtls_test OBJECT ${MBEDTLS_TEST_FILES})
+    target_include_directories(mbedtls_test
+        PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/tests/include
+        PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include
+        PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/library)
```

Comment:
This doesn't look right. Programs under `programs` should not include any header from `library`. Some tests do (“invasive” tests), and it would be too much hassle to have separate build scripts for those so we just use `-I library` to build tests. But if a sample program requires `-I library`, something's wrong.

Reply:
Those include paths are to compile the common test code only (thus currently only helpers.c and random.c). They are not used to compile the C files under programs. I've double checked this with `make --trace` and the only include path to compile C files under programs is `include`.

Reply:
Oh, I see. I hadn't tested, and I'd misunderstood the code. Since this file is likely to be maintained by people who don't know cmake, can you please add a comment? Like a comment just before `if(ENABLE_TESTING OR ENABLE_PROGRAMS)` stating that this paragraph is about compiling `test/src/*` and clarifying what `mbedtls_test` is (I would have naively assumed that it builds a `libmbedtls_test.(a|so)` but that's no it)?

Reply:
Sure, always happy to add comments. Done in an additional commit.